improver/lightning.py

```diff
@@ -204,3 +205,215 @@ def latitude_to_threshold(
+        Separates CAPE, LI, PW, CIN, and APCP cubes and checks that the following
+        match: forecast_reference_time, spatial coords, time-bound interval and
+        that CAPE time is at the lower bound of precipitation accumulation time.
+        """
+
+        output_cubes = iris.cube.CubeList()
+        input_names = [
+            "atmosphere_specific_convective_available_potential_energy",
+            "temperature_difference_between_ambient_air_and_air_lifted_adiabatically",
+            "precipitable_water",
+            "atmosphere_specific_convective_inhibition",
+            "precipitation_amount",
+        ]
+
+        for input_name in input_names:
+            output_cubes.append(self._extract_input(cubes, input_name))
+
+        cape, liftidx, pwat, cin, apcp = output_cubes
+
+        (apcp_time,) = list(apcp.coord("time").cells())
+        for cube in [cape, liftidx, pwat, cin]:
+            (cube_time,) = list(cube.coord("time").cells())
+            (apcp_time,) = list(apcp.coord("time").cells())
+            if cube_time.point != apcp_time.bound[0]:
+                raise ValueError(
+                    f"The {cube.name()} time point ({cube_time.point}) should be valid at the "
+                    f"precipitation_accumulation cube lower bound ({apcp_time.bound[0]})."
+                )
+        if np.diff(apcp_time.bound) not in [timedelta(hours=3)]:
+            raise ValueError(
+                f"Precipitation_accumulation cube time window must be three hours, "
+                f"not {np.diff(apcp_time.bound)}."
+            )
+        # Following time/space checks should be made for all diagnostics
+        for cube in [cape, liftidx, pwat, cin]:
+            if cube.coord("forecast_reference_time") != apcp.coord(
+                "forecast_reference_time"
+            ):
+                raise ValueError(
+                    f"{cube.name()} and {apcp.name()} do not have the same forecast reference time"
+                )
+        for cube in [cape, liftidx, pwat, cin]:
+            if not spatial_coords_match([cube, apcp]):
+                raise ValueError(
+                    f"{cube.name()} and {apcp.name()} do not have the same spatial "
+                    f"coordinates"
+                )
+
+        return cape, liftidx, pwat, cin, apcp
+
+    def process(self, cubes: CubeList, model_id_attr: str = None) -> Cube:
+        """
+        From the supplied CAPE, Lifted Index, Precipitable Water, CIN, and 3-hr Accumulated
+        Precipitation cubes, calculate a probability of lightning cube.
+
+        Args:
+            cubes:
+                Cubes of CAPE, Lifted Index, Precipitable Water, CIN, and 3-hr Accumulated
+                Precipitation cubes.
+            model_id_attr:
+                The name of the dataset attribute to be used to identify the source
+                model when blending data from different models.
+
+        Returns:
+            Cube of lightning data
+
+        Raises:
```

Comment:
A pure stylistic thing is that we would usually put the raises bit of the doc-string into the method that does the raising, so in this case, the `_get_inputs` doc-string. However, as it's a private method you might be doing things a bit differently, therefore feel free to entirely ignore me if you wish.

Reply:
We modeled our `LightningMultivariateProbability_USAF2024` plug-in off of the existing `LightningFromCapePrecip` plug-in, which has the raise doc-strings in a similar location.  To be consistent with the existing module, we're going to leave it as is.